doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,1057 @@
+
+Portfast command will be disabled when the spanning-tree mode is set to `mstp` and Edge Port command will not function under `pvst` spanning-tree mode.
+
+Please note that the Edge Port functionality is equivalent to Portfast, with the only difference being the terminology and command usage based on the spanning-tree mode. The configuration for Edge ports will be stored in the portfast attribute in the database and the backend functionality remains the same as portfast.
+
+## Uplink Fast
+MSTP standard does not support uplink fast so uplink fast functionality will be disable for MSTP.
+
+
+#  Sequence Diagrams
+## MSTP global enable
+
+Pre-Configuring VLAN-to-instance will be mapped to IST instance before actually creating VLAN to avoid convergence churn.
+
+![MSTP Global Enable](images/MSTP_global_enable.png)
+
+## MSTP global disable
+![MSTP Global Disable](images/MSTP_global_disable.png)
+
+## MSTP region name/version change
+![MSTP Region Config](images/MSTP_region_config.png)
+
+## Instance creation
+![MSTP Instance Create](images/MSTP_instance_create.png)
+
+## Instance deletion
+![MSTP Instance Del](images/MSTP_instance_del.png)
+
+## Add VLAN to instance
+![MSTP VLAN Add](images/MSTP_vlan_add.png)
+
+## Del VLAN from instance
+![MSTP VLAN Del](images/MSTP_vlan_del.png)
+
+## Add VLAN member
+![MSTP VLAN Member Add](images/MSTP_vlan_member_add.png)
+
+## Del VLAN member
+![MSTP VLAN Member Del](images/MSTP_vlan_member_del.png)
+
+Update port-state, topology change and instance-interface events remain the same as depicted in [Sequence Diagrams of PVST](https://github.com/sandeep-kulambi/SONiC/blob/631ab18211e7e396b138ace561b7a04e7f7b49a1/doc/stp/SONiC_PVST_HLD.md#4-flow-diagrams).
+
+# Configuration Commands
+Following configuration commands will be provided for configuration of MSTP:
+## Global Level
+- **config spanning_tree {mst} {enable|disable}**
+  - Enables or disables mstp at global level on all ports of the switch.
+  - Only one mode of STP can be enabled at a time.
+- **config spanning_tree mst max_hops \<max-hops-value\>**
+  - Specify the number of maximum hops before the BPDU is discarded inside a region.
+  - max-hops-value: Default: 20, range: 1-40
+- **config spanning_tree mst hello \<hello-value\>**
+  - Specify configuring hello interval in sec for transmission of BPDUs. 
+  Default: 2,  range 1-10
+- **config spanning_tree mst max_age \<max_age-value\>**
+  - Specify configuring maximum time to listen for root bridge in seconds.
+  Default: 2,  range 6-40
+- **config spanning_tree mst port_priority \<port_priority-value\>**
+  - Specify configuring the port level priority for root bridge in seconds.
+  Default: 128,  range 0-240 
+- **config spanning_tree mst cost \<cost-value\>**
+  - Specify configuring the port level priority for root bridge in seconds.
```

Comment:
Does this description needs an update to match the command? And path cost configuration should be supported only per interface or global?

Reply:
Path cost works for both cases per interface and global. These are two separate set of commands for each case.